docs/modules/SDDs/pages/0022-managed-services-in-cluster.adoc

```diff
@@ -0,0 +1,158 @@
+
+The K8up objects are usually needed per service instance.
+
+NOTE: We still need to figure out how these K8up objects are generated. Do we need a special controller for that?
+
+When the application specific Kubernetes Operator already brings native Backup/Restore functionality which can't be easily integrated into K8up, it's to be considered to use this one.
+
+==== Monitoring and Alerting Integration
+
+NOTE: This section is still subject to change, depending on how the main monitoring concept will actually look like.
+
+Each application service should bring it's own `ServiceMonitor`, `PodMonitor` and `PrometheusRule` definitions for the Prometheus Operator. These objects are distributed by the Commodore Component which manages the Kubernetes Operaotor. 
+
+If there are any Grafana Dashboards available, they should also be distributed by the Commodore Component.
+
+=== Risks and Mitigations
+
+==== Operator Updates and Upgrades
+
+There is a risk when upgrading a Kubernetes Operator that the service instance could get restarted, redeployed, deleted, destroyed, or anything else you could imagine as the Kubernetes Operator is the responsible tool taking care of the application service.
+
+To mitigate this a defined test procedure is needed which makes sure that Kubernetes Operator upgrades are safe. This also includes figuring out if objects have changed and are still compatible.
+
+It could also be that object conversion are needed befor a Kubernetes Operator can be upgraded which would need additional tooling.
+
+==== Version mismatch
+
+As the service instance object is managed in a different way and outside of the reach of the Commodore Component of the Kubernetes Operator, it could happen that the instance object requests a version which isn't supported by the Kubernetes Operator. Or that after an upgrade the version isn't support anymore.
+
+It's therefore important to keep track of the service instances and their versions and keep them in sync and comparison to the Kubernetes Operator version. This is something which might be implemented done via the Lieutenant API and inventory.
+
+== Drawbacks
+
+=== Service instantiation
+
+With this approach Project Syn currently only handles the Kubernetes Operator itself including some plumbing, but it doesn't handle the actual service instantiation.
+
+== Alternatives
+
+=== Deployment using Helm
+
+Application service instances could be deployed by using Helm charts, instead of using Kubernetes Operators. This way many advanced functionality and domain knowledge would be missing and would have to be implemented otherwise as deployed Helm Charts are static. That means that no active component takes care of the deployed service (f.e. initialization or reconciliation) and also day-2 operations tasks like upgrades would have to be carried out manually.
+
+=== Deployment using Commodore Components
```

Comment:
I see this as just one way on how customers can instantiate their services.

* Do it manually (discouraged by why not)
* Using any automation/GitOps setup the customer has in place and is familiar with
* Use of Project Syn (customer managed)
* Use of Project Syn (VSHN managed)

How about writing a Jsonnet library? We then can reuse them as we need (render within a commodore component, render OpenShift templates, Helm charts etc.)

Reply:
Not sure what I should do with this comment. This section talks about alternatives. What do you feel isn't clear or should be updated?

Reply:
Probably I was just musing head. This might become relevant when we implement his.